app/root.tsx

```diff
@@ -64,25 +79,25 @@ export function Layout({ children }: React.PropsWithChildren) {
 
 function ContentWrapper({ children }: React.PropsWithChildren) {
     return (
-        <EcomAPIContextProvider>
-            <CartOpenContextProvider>
-                <SiteWrapper>{children}</SiteWrapper>
-            </CartOpenContextProvider>
-        </EcomAPIContextProvider>
+        <CartOpenContextProvider>
+            <SiteWrapper>{children}</SiteWrapper>
+        </CartOpenContextProvider>
     );
 }
 
 export default function App() {
-    const data = useLoaderData<typeof loader>();
+    const { ENV, wixEcomTokens } = useLoaderData<typeof loader>();
 
     if (typeof window !== 'undefined' && typeof window.ENV === 'undefined') {
-        window.ENV = data.ENV;
+        window.ENV = ENV;
     }
 
     return (
-        <ContentWrapper>
-            <Outlet />
-        </ContentWrapper>
+        <EcomAPIContextProvider tokens={wixEcomTokens}>
+            <ContentWrapper>
```

Comment:
can't we pass `tokens` inside the `ContentWrapper`?
I don't like the idea that now `CartOpenContextProvider` is inside the `ContentWrapper`, but `EcomAPIContextProvider` is here... if there was a new context provider I wouldn't know where and what to wrap with it :)

Reply:
fixed